### ISSUE TITLE
Assign values generated or calculated by a query ​​to an entity

### ISSUE THREAD

**Issue by localnet** [B4nan/mikro-orm#40]
I'm not sure if this feature is out of ambit of a ORM. Feel free of close the issue directly, if so.

I have a entity:
```typescript
@Entity({ collection: "location" })
export class LocationModel implements EntityLocation {
    @PrimaryKey()
    public _id!: ObjectId;

    @Property()
    public geometry!: EntityGeometry;

    @ManyToOne({ entity: () => BusinessModel })
    public business!: BusinessModel;

    public distance?: number;
}

export interface LocationModel extends IEntity<string> { }
```

And a query, MongoDB aggregation, that create a new value calculated:
```typescript
public async findNear(coordinates: number[]): Promise<LocationModel[]> {
    const entities: EntityLocation[] = await this.locationRepository.aggregate([
        {
            $geoNear: {
                distanceField: "distance",
                distanceMultiplier: 0.001,
                near: { coordinates, type: "Point" },
                spherical: true,
            },
        },
    ]);

    return entities.map((entity) => this.locationRepository.create(entity));
}
```

When converting the query result to entity, the value "distance" is ignored and I have to handle it manually. Can be possible implement something to handle it automatically, something like:
```typescript
@Entity({ collection: "location" })
export class LocationModel implements EntityLocation {
    @PrimaryKey()
    public _id!: ObjectId;

    @Property()
    public geometry!: EntityGeometry;

    @ManyToOne({ entity: () => BusinessModel })
    public business!: BusinessModel;

    @Property({ persist: false })
    public distance?: number;
}

export interface LocationModel extends IEntity<string> { }
```

**Comment by B4nan** [comment]
`EntityManager.create()` currently ignores properties that are not marked via decorators. You could create reference instead and use assign to include them:

    return entities.map((data) => {
      const entity = this.locationRepository.getReference(data._id);
      entity.assign(data);
      return entity;
    });

I will add bool parameter to create method so you can achieve it via single command, it makes sense to allow this.

**Comment by B4nan** [comment]
Or do you think its better to implement `@Property({ persist: false })`? Should be pretty easy. 

It might be better as you could whitelist properties instead of allowing anything via one boolean.

**Comment by localnet** [comment]
I am sure you have more experience with ORMs. If you see some advantage to my proposition, perfect I will use it, else I don't want add maintenance complexity with features used only by me.

With the information of your previous response I can do a generic function to solve this, and can be enougth:
```typescript
public static toModel<T>(entities: BaseEntity[], repository: EntityRepository<IEntityType<T>>): Array<IEntityType<T>> {
    return entities.map((entity) => {
        const location: IEntityType<T> = repository.getReference(entity.id);

        location.assign(entity);

        return location;
    });
}
```

**Comment by localnet** [comment]
Or maybe simply if `assign()´ return the entity, so that it can be done:
```typescript
return entities.map((entity) => this.locationRepository.getReference(entity.id).assign(entity));
```

instead of:
```typescript
return entities.map((entity) => {
    const location: LocationModel = this.locationRepository.create(entity);
    location.assign(entity);
    return location;
});
```

But I prefer you choose, because you know the code better and I don't finish to understand when to use `EntityRepository.create()` or `new Entity()` and `Entity.assign()` (Surely I will ask you at stackoverflow about this).

**Comment by B4nan** [comment]
Fluent interface to entity.assign() also came to my mind. I will probably do all 3 things, fluent interface for `entity.assign()`, boolean toggle for `EM.create()` to allow extra properties, and `@Propery({ persist: false: })` so one can whitelist only specific properties that are allowed. 

I expect to have some time for this later this week, will ping you here once its published. Or feel free to send PR, 1. and 2. are really small changes. 3. will be probably more challenging and will require some additional tests, I can handle that myself. 

Btw there is also `EM.merge()`, that behaves similarly to `EM.create()` :] I will definitely write something about this into docs later, but you can check tests to see how those things differ in the meanwhile. Currently I am trying to gather some feedback on reddit and once I have enough, I will create FAQ page. I also published an article about MikroORM: https://medium.com/dailyjs/introducing-mikro-orm-typescript-data-mapper-orm-with-identity-map-9ba58d049e02

**Comment by B4nan** [comment]
Shadow properties implemented in 6062118aee036037786f1c06f02cfa20eb124f2e

I saw you were forking the repo, are you up to implement 1. and 2.? If not I will do that myself, but will be glad for another collaborator :)

**Comment by localnet** [comment]
No, I was trying the MongoDB transactions. I was too ambitious. I can't see how to handle the possible nested transactions.

**Comment by B4nan** [comment]
Released v2.2 that includes shadow properties.

**Comment by localnet** [comment]
Great, I expect to have some time to try it later this week.

Thanks.
